### PR TITLE
Prescriptions and MAR: Disallow users to perform edits on discharged encounters

### LINKED ISSUE #2105: Restrict all medicine prescription related activity for discharge patients
**Describe the bug**

When we discharge a patient in care, currently we can do all the below functions:
- [x] We can select individual medicine and administer it
image
- [x] We can edit and discontinue all existing medicines by clicking on the individual medicine name
- [x] users can archive any already administrated status

image

**To Reproduce**
1. Go to the medicine section of discharge patients who have existing medicine
2. execute all above mentioned activities 
3. See error

**Expected behavior**

All medicine-related administration and editing should be restricted after the patient's discharge

### LINKED ISSUE #7755: Nurse user type without home facility is able to edit prescriptions
**Describe the bug**
Nurse user id without Homefacility shouldn't be able to edit the prescription

**To Reproduce**

1. Go to patient dashboard
2. Click on medication and edit prescription

**Expected behavior**
When click on 'edit prescription', Show some error message saying no permission to do this task without a homefacility linked

### PR DESCRIPTION
**Pull request by rithviknishad**
## Proposed Changes

- Disallow actions to Prescription and Medicine Administration Records for discharged encounters.
- Merge prescription related tests to a single file and have dedicated test file for medicine administrations

### Associated Issue

- [x] Fixes https://github.com/coronasafe/care_fe/issues/7755
- [x] Fixes #2105

## Merge Checklist

- [x] Tests added/fixed
- [x] Linting Complete

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

**Comment by rithviknishad** [comment]
Waiting for confirmation regd. whether all remote users needs to be restricted or just remote nurses.

**Comment by rithviknishad** [comment]
Confirmed. Doctors and Nurses of home facility only must be able to edit consultation related data. Other than doctor notes.

**Review by sainak — APPROVED** [review]

**Review by shivankacker — APPROVED** [review]

**Review by nihal467 — CHANGES_REQUESTED** [review]
- admin should be able to edit it irrespective of home facility privileges
- non-home facility staff is able to create medicine. block that as well in frontend as well backend

**Comment by nihal467** [comment]
@rithviknishad any update on this, is it good for review

**Review by sainak — APPROVED** [review]

File: care/facility/api/viewsets/prescription.py [hunk 1]
```diff
@@ -76,6 +77,10 @@ def get_queryset(self):
     @extend_schema(tags=["prescription_administration"])
     @action(methods=["POST"], detail=True)
     def archive(self, request, *args, **kwargs):
+        if self.get_consultation_obj().discharge_date:
```
Comment (vigneshhari) [inline care/facility/api/viewsets/prescription.py:82]: This needs to be added in discontinue api as well

**Review by vigneshhari — CHANGES_REQUESTED** [review]

**Comment by gigincg** [comment]
@rithviknishad Waiting for your changes

**Review by sainak — APPROVED** [review]

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**